tests/driver.rs

```diff
@@ -90,6 +91,28 @@ fn too_many_submissions() {
     });
 }
 
+#[test]
+fn completion_overflow() {
+    let spawn_cnt = 100;
+    let squeue_entries = 2;
+    let cqueue_entries = 2 * squeue_entries;
```

Comment:
How about adding a second timeout so when this does hang, it fails relatively quickly. I found this worked well

```rust
    std::thread::spawn(|| {
        thread::sleep(time::Duration::from_millis(1000));
        eprintln!("Test failed, timeout reached.");
        process::exit(1);
    });
```

Reply:
I don't think that using `std::thread` and `process::exit` is a good way of doing this. If we need to do this with a timeout, I'd rather we wrap an async block in `tokio::time::timeout`.

Generally for things like this, I prefer to move the main logic off to a spawned task and to have the entry task run `tokio::task::yield_now().await` a bunch of times in a row before checking if we are finished or not and exiting. This allows us to be bounded by how many cycles through the scheduler we take rather than by time elapsed, and is far more deterministic in CI.

Reply:
Am looking for an example of this in a test. Would like to get a version of this unit test committed this morning with a timeout in it now that io-uring 0.5.8 is available - and I'm preparing a commit to bump the io-uring dependency in tokio-uring's Cargo.toml.

Reply:
I couldn't get any option like that to work. It's always possible that when it's hung, it is really blocking the entire thread.